Code/src/main/java/nl/utwente/group10/haskell/type/Type.java

```diff
@@ -2,45 +2,29 @@
 
 import nl.utwente.group10.haskell.HaskellObject;
 
+import java.util.logging.Logger;
+
 /**
  * Abstract class for Haskell types. Provides an interface for common methods.
  *
  * Types should be immutable. Also, a Type subclass instance that represents the same in Haskell as another instance of
  * the same class should be equal, that is {@code this.equals(that)} should return {@code true}.
  */
-public abstract class Type extends HaskellObject implements Comparable<Type> {
+public abstract class Type extends HaskellObject {
```

Comment:
The JavaDoc above might no longer be true: unify does some horrible, horrible things to types, which probably makes them no longer immutable, and the instance/equals stuff I'm not sure about either.